### PR TITLE
Add troubleshooting section for tpus

### LINKED ISSUE #8128: Create a Troubleshooting section for TPU Support docs
## 📚 Documentation

Create a Troubleshooting section for TPU Support docs

Some pointers:

- Missing XLA Configuration
- How to clear up the programs using TPUs in the background
- Replication issue
- Unsupported datatype transfer to TPU issue
- Using `tpu_spawn_debug` plugin
- Troubleshooting for TPU Pod training
...

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes #8128 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — COMMENTED** [review]
So far, neat !

File: docs/source/advanced/tpu.rst [hunk 1]
```diff
@@ -294,6 +294,78 @@ has more detailed information on how PyTorch code can be optimized for TPU. In p
...
+
+.. code-block::
+
+    File "/usr/local/lib/python3.6/dist-packages/torch_xla/core/xla_model.py", line 200, in set_replication
+        replication_devices = xla_replication_devices(devices)
+    File "/usr/local/lib/python3.6/dist-packages/torch_xla/core/xla_model.py", line 187, in xla_replication_devices
+        .format(len(local_devices), len(kind_devices)))
+    RuntimeError: Cannot replicate if number of devices (1) is different from 8
+
+This error is raised when the XLA device is called outside the spawn process. Don't call `xm.xla_device()` outside spawn.
```
Comment (awaelchli) [inline docs/source/advanced/tpu.rst:348]:
```suggestion
This error is raised when the XLA device is called outside the spawn process. Don't call ``xm.xla_device()`` outside spawn.
```

**Review by awaelchli — APPROVED** [review]
nice!

**Review by tchaton — APPROVED** [review]
Looks great !

File: docs/source/advanced/tpu.rst [hunk 2]
```diff
@@ -294,6 +294,119 @@ has more detailed information on how PyTorch code can be optimized for TPU. In p
...
+        Percentiles: 1%=001ms32.778us; 5%=001ms61.283us; 10%=001ms79.236us; 20%=001ms110.973us; 50%=001ms228.773us; 80%=001ms339.183us; 90%=001ms434.305us; 95%=002ms921.063us; 99%=21s102ms853.173us
+
+
+A lot of PyTorch operations aren't lowered to XLA, which could lead to significant slowdown of the training process.
+These operations are moved to the CPU memory and evaluated, and then the results are transfered back to the XLA device(s).
+By using the `tpu_spawn_debug` plugin, users could create a metrics report to diagnose issues.
+
+The report includes things like (`XLA Reference <https://github.com/pytorch/xla/blob/master/TROUBLESHOOTING.md#troubleshooting>`_):
+
+* how many time we issue XLA compilations and time spent on issuing.
```
Comment (awaelchli) [inline docs/source/advanced/tpu.rst:390]:
```suggestion
* how many times we issue XLA compilations and time spent on issuing.
```

**Review by awaelchli — APPROVED** [review]

File: docs/source/advanced/tpu.rst [hunk 3]
```diff
@@ -294,6 +294,119 @@ has more detailed information on how PyTorch code can be optimized for TPU. In p
...
+    export TPU_IP_ADDRESS=10.39.209.42  # You could get the IP Address on GCP TPUs section
+    export XRT_TPU_CONFIG="tpu_worker;0;$TPU_IP_ADDRESS:8470"
+
+- **How to clear up the programs using TPUs in the background**
+
+.. code-block:: bash
+
+    lsof -w /lib/libtpu.so | grep "python" |  awk '{print $2}' | xargs -r kill -9
+
+Sometimes, there would still be old processes running on the TPUs, which would make the TPUs unavailable to use. You could use the above command in the terminal to clear up the running programs.
```
Comment (carmocca) [inline docs/source/advanced/tpu.rst:336]: nits:

```suggestion
Sometimes, there can still be old programs running on the TPUs, which would make the TPUs unavailable to use. You could use the above command in the terminal to kill the running processes.
```

File: docs/source/advanced/tpu.rst [hunk 4]
```diff
@@ -294,6 +294,119 @@ has more detailed information on how PyTorch code can be optimized for TPU. In p
...
+
+.. code-block::
+
+    File "/usr/local/lib/python3.6/dist-packages/torch_xla/core/xla_model.py", line 200, in set_replication
+        replication_devices = xla_replication_devices(devices)
+    File "/usr/local/lib/python3.6/dist-packages/torch_xla/core/xla_model.py", line 187, in xla_replication_devices
+        .format(len(local_devices), len(kind_devices)))
+    RuntimeError: Cannot replicate if number of devices (1) is different from 8
+
+This error is raised when the XLA device is called outside the spawn process. Don't call ``xm.xla_device()`` outside spawn.
```
Comment (carmocca) [inline docs/source/advanced/tpu.rst:348]: Can you elaborate on what this means for the naive Lightning user who probably doesn't know how spawn works or what it is?

File: docs/source/advanced/tpu.rst [hunk 5]
```diff
@@ -294,6 +294,119 @@ has more detailed information on how PyTorch code can be optimized for TPU. In p
...
+
+.. code-block::
+
+    File "/usr/local/lib/python3.8/dist-packages/torch_xla/utils/utils.py", line 205, in _for_each_instance_rewrite
+        v = _for_each_instance_rewrite(result.__dict__[k], select_fn, fn, rwmap)
+    File "/usr/local/lib/python3.8/dist-packages/torch_xla/utils/utils.py", line 206, in _for_each_instance_rewrite
+        result.__dict__[k] = v
+    TypeError: 'mappingproxy' object does not support item assignment
+
+PyTorch XLA only supports Tensor objects for CPU to TPU data transfer.
```
Comment (carmocca) [inline docs/source/advanced/tpu.rst:360]: This explains the error, but when would this happen for your average user?

**Review by carmocca — COMMENTED** [review]
Love these!

File: docs/source/advanced/tpu.rst (hunk 4 above)
Comment (kaushikb11) [reply docs/source/advanced/tpu.rst]: Updated!

File: docs/source/advanced/tpu.rst (hunk 5 above)
Comment (kaushikb11) [reply docs/source/advanced/tpu.rst:361]: If the User is trying to send some non-tensor objects through the DataLoader or during saving states.

Comment (carmocca) [reply docs/source/advanced/tpu.rst:361]: Might be worth adding

**Review by carmocca — APPROVED** [review]

**Pull request merged**